File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_table = {}
        for i,n in enumerate(nums):
            # if number in hash table, return indices
            if n in hash_table: return [hash_table.get(n), i]
            hash_table[target-n] = i

File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int, int> map;
        for (int n : nums) {
            if (map.count(n)) {
                return true;
            } else {
                map[n] = 1;
            }
        }
        return false;
    }
};
